spec/request/user/create_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'POST #create sign_up', type: :request do  
+  let!(:user_params) { { user: FactoryBot.attributes_for(:user) } }
+  subject do
+    post '/api/v1/users', params: user_params, as: :json
+  end
+
+  context 'when valid' do
+    it 'does get a sucessful answer' do
+      subject
+      expect(response).to be_successful
+    end
+
+    it 'does create a new user' do
+      expect { subject }.to change { User.count }.by(1)
+    end
+
+    it 'the attributes from params persist on db' do
+      subject
+      expect(response.body).to include_json(
+        user: {
+          email: User.last.email,
+          gender: User.last.gender,
+          full_name: User.last.full_name
```

Comment:
Maybe in this case that you use `User.last` several times, you can define `let(:user) { User.last }` and use it here like:
```
email: user.email,
gender: user.gender,
full_name: user.full_name
```

Reply:
Where in particular should I define the User? 
Because due to the test, the user is created during the subject, then i should call the let!(:user) after the subject has been called so i can reference to the -last user- indeed, the thing is it wont allow me to call let inside the `it` block.

Reply:
You can define the `let` before the `context` starts. For example:
```
RSpec.describe 'POST #create sign_up', type: :request do  
  let!(:user_params) { { user: FactoryBot.attributes_for(:user) } }
  let(:user)         { User.last }
  subject do
    post '/api/v1/users', params: user_params, as: :json
  end
```
If you don't use the `!` (just `let`), the `User.last` isn't going to be called until you call `user`. So if you call `user` inside the `expect` the user will be already created.
You can see some explanation and examples [here](http://www.betterspecs.org/es/#let) and [here](https://mixandgo.com/learn/let-vs-instance-variables-in-rspec).

Reply:
Great, thanks!